### PR TITLE
added nitro check to prefix provider and pod controller

### PR DESCRIPTION
**Pull request by jiechen0826**
*Issue #, if available:*

*Description of changes:*
- added nitro check to prefix provider

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/provider/prefix/provider.go [hunk 1]
```diff
@@ -334,8 +334,12 @@ func (p *ipv4PrefixProvider) GetPool(nodeName string) (pool.Pool, bool) {
...
-	//TODO add check for Nitro instances
-	if instance.Os() == config.OSWindows {
+	limits, found := vpc.Limits[instance.Type()]
+	if !found {
+		return false
+	}
+
+	if instance.Os() == config.OSWindows && (limits.IsBareMetal || limits.Hypervisor == "nitro") {
 		return true
 	}
```
Comment (haouc) [inline pkg/provider/prefix/provider.go:345]: Can you emit an event for cases of not found and not supported?

Comment (jiechen0826) [reply pkg/provider/prefix/provider.go:355]: Yea sure, I've added node events for both cases.

**Comment by jiechen0826** [comment]
I'm checking why the unit test failed here, and it looks like it's a connectivity issue with Github. The unit tests are passing locally. I can force update to trigger the check again.

File: pkg/provider/prefix/provider.go (hunk 1 above)
Comment (haouc) [reply pkg/provider/prefix/provider.go:355]: Thanks, @jiechen0826. Please also make sure these user faced messages are tested (positive and negative).

**Review by haouc — APPROVED** [review]
lgtm, thanks
